### ISSUE TITLE
Unable to sort while using relay connection pagination.

### ISSUE THREAD

**Issue by lynndylanhurley** [rmosolgo/graphql-ruby#1353]
The following error occurs when trying to paginate with an `order` applied:

~~~
     ActiveRecord::StatementInvalid:
       PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
       LINE 1: ...nator_id = 1 OR assignments.user_id = 1) ORDER BY title OFFS...
                                                                    ^
       : SELECT COUNT(DISTINCT count_column) FROM (SELECT DISTINCT "matters"."id" AS count_column FROM "matters" LEFT OUTER JOIN "assignments"
ON "assignments"."matter_id" = "matters"."id" AND "assignments"."deleted_at" IS NULL WHERE (matters.originator_id = 1 OR assignments.user_id =
1) ORDER BY title OFFSET $1) subquery_for_count
~~~

The issue can be resolved by unscoping the `order` before performing the `count` on [this line](https://github.com/rmosolgo/graphql-ruby/blame/f6aab69afe53505f0bb4e0bf8fbae9e6152e4fd2/lib/graphql/relay/relation_connection.rb#L127).

This works:
~~~ruby
      # If a relation contains a `.group` clause, a `.count` will return a Hash.
      def relation_count(relation)
        count_or_hash = if(defined?(ActiveRecord::Relation) && relation.is_a?(ActiveRecord::Relation))
          # unscope the :order before performing the `count`
          relation.unscope(:order).count(:all)
        else # eg, Sequel::Dataset, don't mess up others
          relation.count
        end
        count_or_hash.is_a?(Integer) ? count_or_hash : count_or_hash.length
      end
~~~

Would you accept a PR with this change?

**Comment by sideshowdave7** [comment]
I second this for performance reasons - sending an ordered relation, especially when ordering by fields from joined tables can be really expensive in MySQL.

**Comment by sideshowdave7** [comment]
I did some more testing and ran into the same error as @lynndylanhurley actually.  I've put in a PR that addresses the issue (supported by Rails 4/5)

**Comment by rmosolgo** [comment]
Fixed by #1911
